safenode/src/domain/storage/disk_backed_record_store.rs

```diff
@@ -246,10 +240,19 @@ impl RecordStore for DiskBackedRecordStore {
         }
     }
 
+    // A backstop replication shall only trigger within pre-defined interval
     fn records(&self) -> Self::RecordsIter<'_> {
+        // Only need to load portion of the records.
```

Comment:
Hmm, this is not the targetted replication. 

So if we're maintaining parity of `RecordStore`, this has to return all records.

So our periodic libp2p backed flow will do just that, and send out everything. (And so will likely be over a larger period).

Then above, we should probably avoid calling `records` to avoid pulling _everything_ in. 

If we can't track only the files/keys we want at a higher level, we may want to try upstreaming some helper fns to libp2p?